File: client-scalaz/src/main/scala/org/openapitools/client/api/WrappedProxyItemResponse.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import WrappedProxyItemResponse._

case class WrappedProxyItemResponse (
  valid: Option[Boolean],
message: Option[String],
version: Option[Double],
serializeNulls: Option[Boolean],
startTimeLog: Option[Long],
errorCode: Option[String],
request: Option[List[NameStringValueResponse]],
returning: Option[String],
`type`: Option[String],
responseCode: Option[Integer],
responseRaw: Option[String],
requestDuration: Option[Long],
requestEndTime: Option[Long],
item: Option[ImageGenerationResponse])

object WrappedProxyItemResponse {
  import DateTimeCodecs._

  implicit val WrappedProxyItemResponseCodecJson: CodecJson[WrappedProxyItemResponse] = CodecJson.derive[WrappedProxyItemResponse]
  implicit val WrappedProxyItemResponseDecoder: EntityDecoder[WrappedProxyItemResponse] = jsonOf[WrappedProxyItemResponse]
  implicit val WrappedProxyItemResponseEncoder: EntityEncoder[WrappedProxyItemResponse] = jsonEncoderOf[WrappedProxyItemResponse]
}

File: client-clojure/src/sirqul_io_t_platform/specs/wrapped_proxy_item_response.clj
(ns sirqul-io-t-platform.specs.wrapped-proxy-item-response
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [sirqul-io-t-platform.specs.name-string-value-response :refer :all]
            [sirqul-io-t-platform.specs.image-generation-response :refer :all]
            )
  (:import (java.io File)))


(def wrapped-proxy-item-response-data
  {
   (ds/opt :valid) boolean?
   (ds/opt :message) string?
   (ds/opt :version) float?
   (ds/opt :serializeNulls) boolean?
   (ds/opt :startTimeLog) int?
   (ds/opt :errorCode) string?
   (ds/opt :request) (s/coll-of name-string-value-response-spec)
   (ds/opt :returning) string?
   (ds/opt :type) string?
   (ds/opt :responseCode) int?
   (ds/opt :responseRaw) string?
   (ds/opt :requestDuration) int?
   (ds/opt :requestEndTime) int?
   (ds/opt :item) image-generation-response-spec
   })

(def wrapped-proxy-item-response-spec
  (ds/spec
    {:name ::wrapped-proxy-item-response
     :spec wrapped-proxy-item-response-data}))
